### PR TITLE
Allow running test-cmd on osx

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

File: hack/test-cmd.sh [hunk 1]
```diff
@@ -310,7 +320,7 @@ runTests() {
   kubectl config set clusters.test-cluster.certificate-authority-data "$cert_data" --set-raw-bytes
   r_writen=$(kubectl config view --raw -o jsonpath='{.clusters[?(@.name == "test-cluster")].cluster.certificate-authority-data}')
 
-  encoded=$(echo -n "$cert_data" | base64 --wrap=0)
+  encoded=$(echo -n "$cert_data" | base64)
```
Comment (ihmccreery) [inline hack/test-cmd.sh]: `--wrap=0` seems important here; why get rid of it?

File: hack/test-cmd.sh [hunk 2]
```diff
@@ -707,13 +717,13 @@ __EOF__
...
   ## kubectl edit can update the image field of a POD. tmp-editor.sh is a fake editor
-  echo -e '#!/bin/bash\nsed -i "s/nginx/gcr.io\/google_containers\/serve_hostname/g" $1' > /tmp/tmp-editor.sh
+  echo -e "#!/bin/bash\n$SED -i \"s/nginx/gcr.io\/google_containers\/serve_hostname/g\" \$1" > /tmp/tmp-editor.sh
   chmod +x /tmp/tmp-editor.sh
   EDITOR=/tmp/tmp-editor.sh ${KUBE_OUTPUT_HOSTBIN}/kubectl edit "${kube_flags[@]}" pods/valid-pod
   # Post-condition: valid-pod POD has image gcr.io/google_containers/serve_hostname
   kube::test::get_object_assert pods "{{range.items}}{{$image_field}}:{{end}}" 'gcr.io/google_containers/serve_hostname:'
   # cleaning
-  rm /tmp/tmp-editor.sh
+  # rm /tmp/tmp-editor.sh
```
Comment (ihmccreery) [inline hack/test-cmd.sh]: ?

Comment (liggitt) [inline hack/test-cmd.sh]: weird, reverted

File: hack/test-cmd.sh (hunk 1 above)
Comment (liggitt) [inline hack/test-cmd.sh]: it's nonstandard, so it breaks on osx, and it's unnecessary, because the go base64 decoder skips newlines

**Comment by ihmccreery** [comment]
LGTM

**Pull request merged**